content/v2/openapi.yml

```diff
@@ -1564,6 +1564,32 @@ paths:
               schema:
                 $ref: '#/components/schemas/Error'
       summary: Check zone distribution
+  '/{account}/zones/{zone}/ns_records':
+    put:
+      description: Updates the zone's NS records
+      parameters:
+        - $ref: '#/components/parameters/Account'
+        - $ref: '#/components/parameters/Zone'
+        - $ref: '#/components/parameters/NsSetIds'
+        - $ref: '#/components/parameters/NsNames'
```

Comment:
How will `NsNames` be used, given that we have `NsSetIds` ?

Reply:
The idea is that you can also provide ['ns1.example.com'] – like the form.

Forced to choose one, I tend to think the NsNames is more valuable to have because, if you're using the API, it is likely easier for you to manage the sets within your own source code. That being said, we'll also need to expand it for glue records (which wouldn't be verified as its not in our currently-laid-out work).

WDYT, remove the NsNames param?

Reply:
I see, I think including both makes sense (for maximum flexibility), and I agree with just implementing support for  NsSetIds if we are constrained by time 🙂